PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/model/PhylogeneticTreeNode.java

```diff
@@ -1,109 +1,240 @@
 package nl.tudelft.pl2016gr2.gui.model;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode;
+
 import java.util.ArrayList;
-import java.util.Objects;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 /**
- * This class is an adapter class which maps the methods of the
- * net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode class to the interface which is needed by
- * our application.
+ * This class is a storage class for a phylogenetic tree node. It retrieves and stores the needed
+ * values of the newick parser {@link TreeNode}.
  *
  * @author Faris
  */
-public class PhylogeneticTreeNode implements IPhylogeneticTreeNode {
+public class PhylogeneticTreeNode implements IPhylogeneticTreeNode, Iterable<PhylogeneticTreeNode> {
+
+  private final String label;
+  private final float weight;
+  private final PhylogeneticTreeNode[] children;
+  private final PhylogeneticTreeNode parent;
+
+  /**
+   * If all of the child nodes of this node are drawn in the top graph.
+   */
+  private BooleanProperty drawnInTop = new SimpleBooleanProperty(false);
 
-  private final net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode node;
+  /**
+   * If all of the child nodes of this node are drawn in the bottom graph.
+   */
+  private BooleanProperty drawnInBottom = new SimpleBooleanProperty(false);
 
   /**
    * Construct a phylogenetic tree node.
    *
-   * @param node the TreeNode of this node.
+   * @param node   the TreeNode of this node.
+   * @param parent the parent of this phylogenetic tree node.
    */
-  public PhylogeneticTreeNode(net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode node) {
-    this.node = node;
+  protected PhylogeneticTreeNode(TreeNode node, PhylogeneticTreeNode parent) {
+    this.label = node.label;
+    this.weight = node.weight;
+    this.parent = parent;
+
+    if (node.numberChildren() == 2) {
+      children = new PhylogeneticTreeNode[2];
+      children[0] = new PhylogeneticTreeNode(node.getChild(0), this);
+      children[1] = new PhylogeneticTreeNode(node.getChild(1), this);
+    } else {
+      children = null;
+    }
```

Comment:
A newick file can have multiple children per node, but a phylogenetic tree can't. Since the class is called `PhylogeneticTreeNode` I think this is clear enough already. I don't think it's really needed, but I will add two line comments.